clang/lib/Lex/HeaderSearch.cpp

```diff
@@ -1458,7 +1458,7 @@ bool HeaderSearch::ShouldEnterIncludeFile(Preprocessor &PP,
   } else {
     // Otherwise, if this is a #include of a file that was previously #import'd
     // or if this is the second #include of a #pragma once file, ignore it.
-    if ((FileInfo.isPragmaOnce || FileInfo.isImport) && !TryEnterImported())
```

Comment:
Technically, `TryEnterImported` can be inlined at another call site but I think it's not worth it and we can keep things the way you have in PR.

Reply:
I feel it will become harder to read. So I didn't inline it. But I move the definition to a closer place.